stl/src/mutex.cpp

```diff
@@ -19,19 +19,13 @@ extern "C" [[noreturn]] _CRTIMP2_PURE void _Thrd_abort(const char* msg) { // abo
     abort();
 }
 
-#if defined(_THREAD_CHECK) || defined(_DEBUG)
-#define _THREAD_CHECKX 1
-#else // defined(_THREAD_CHECK) || defined(_DEBUG)
-#define _THREAD_CHECKX 0
-#endif // defined(_THREAD_CHECK) || defined(_DEBUG)
-
-#if _THREAD_CHECKX
+#ifdef _DEBUG
 #define _THREAD_QUOTX(x)          #x
 #define _THREAD_QUOT(x)           _THREAD_QUOTX(x)
 #define _THREAD_ASSERT(expr, msg) ((expr) ? (void) 0 : _Thrd_abort(__FILE__ "(" _THREAD_QUOT(__LINE__) "): " msg))
-#else // _THREAD_CHECKX
+#else // defined(_DEBUG)

```

Comment:
This conflicts with #3900 , maybe easier to postpone this part to avoid conflicts?

Reply:
That's a really simple merge conflict to resolve, but thanks for the heads up. I'll remember to upgrade this one to arrow comments.